src/main.cc

```diff
@@ -1052,6 +1055,13 @@ int main(int argc, char** argv)
 			reporter->Warning("event handler never invoked: %s", handler.c_str());
 		}
 
+	// Enable LeakSanitizer before zeek_init() and even before executing
+	// top-level statements.  Even though it's not bad if a leak happens only
+	// once at initialization, we have to assume that script-layer code causing
+	// such a leak can be placed in any arbitrary event handler and potentially
+	// cause more severe problems.
+	ZEEK_LSAN_ENABLE();
```

Comment:
In theory, that generally sounds fine, but don't think it should hold up the PR since this location is already better than before, should give coverage for the full btest suite, and once this framework is in place someone else can even take on the task of moving it up higher and addressing the leaks the come with that (I'm a bit burnt out on tracking leaks for now).

I'm also more nuanced about the all-leaks-are bugs stance -- it actually seems somewhat of a defect or non-feature for a program to slow down its own shutdown to free complex data structures allocated at initialization-time when the OS is just going to reclaim that all more quickly anyway.  But yeah, I'd also suggest just generally taking the Good Practice approach of freeing everything as it can reveal other unintentional bugs, too, and can always have a "fast exit" mode that doesn't bother doing that if shutdown speed ever was a concern for a particular program.